src/main/java/it/pagopa/pn/deliverypush/service/mapper/TimelineMapperAfterFix.java

```diff
@@ -0,0 +1,113 @@
+                            log.debug("MAP TIMESTAMP: elem category {}, elem previous timestamp {}, elem new timestamp {} ", result.getCategory(), result.getTimestamp(), endAnalogWorkflowBusinessDate);
+                            result.setTimestamp(endAnalogWorkflowBusinessDate);
+                        }
+                    }
+                }
+                case ANALOG_SUCCESS_WORKFLOW -> {
+                    Instant endAnalogWorkflowBusinessDate = computeEndAnalogWorkflowBusinessData((RecipientRelatedTimelineElementDetails) result.getDetails(), timelineElementInternalSet, result.getIun());
+                    if (endAnalogWorkflowBusinessDate != null) {
+                        log.debug("MAP TIMESTAMP: elem category {}, elem previous timestamp {}, elem new timestamp {} ", result.getCategory(), result.getTimestamp(), endAnalogWorkflowBusinessDate);
+                        result.setTimestamp(endAnalogWorkflowBusinessDate);
+                    } else {
+                        log.error("SEARCH LAST SEND_ANALOG_FEEDBACK DETAILS NULL element {}", result);
+                        throw new PnInternalException("SEND_ANALOG_FEEDBACK NOT PRESENT, ERROR IN MAPPING", PnDeliveryPushExceptionCodes.ERROR_CODE_DELIVERYPUSH_TIMELINE_ELEMENT_NOT_PRESENT);
+                    }
+                }
+                case COMPLETELY_UNREACHABLE_CREATION_REQUEST, COMPLETELY_UNREACHABLE -> {
+                    Instant analogFailureWorkflowDate = getAnalogFailureWorkflowDate((RecipientRelatedTimelineElementDetails) result.getDetails(), timelineElementInternalSet, result.getIun());
+                    if (analogFailureWorkflowDate != null)
+                        result.setTimestamp(analogFailureWorkflowDate);
+                    else
+                        throw new PnInternalException("ANALOG_FAILURE_WORKFLOW NOT PRESENT, ERROR IN MAPPING", PnDeliveryPushExceptionCodes.ERROR_CODE_DELIVERYPUSH_TIMELINE_ELEMENT_NOT_PRESENT);
+                }
+                case REFINEMENT -> {
+                    ScheduleRefinementDetailsInt details = findScheduleRefinementDetails((RecipientRelatedTimelineElementDetails) result.getDetails(), timelineElementInternalSet);
+                    if (details != null) {
+                        log.debug("MAP TIMESTAMP: elem category {}, elem previous timestamp {}, elem new timestamp {}", result.getCategory(), result.getTimestamp(), details.getSchedulingDate());
+                        result.setTimestamp(details.getSchedulingDate());
+                    } else {
+                        throw new PnInternalException("SCHEDULE_REFINEMENT NOT PRESENT, ERROR IN MAPPING", PnDeliveryPushExceptionCodes.ERROR_CODE_DELIVERYPUSH_TIMELINE_ELEMENT_NOT_PRESENT);
+                    }
+                }
+                case SEND_DIGITAL_DOMICILE -> {
+                    // allo scopo di ottenere il timestamp di AAR_GEN per impostare il timestamp di SEND_DIGITAL_DOMICILE, per far si
```

Comment:
Questo pezzo, lo metterei in un metodo, spostato a fattor comune nella classe astratta richiamato poi da entrambe le implementazioni

Reply:
Anche per il REFINEMENT probabilmente può essere fatta la stessa cosa